### DISCUSSION TITLE
Why did installing weasyprint upgrade mariadb?

### DISCUSSION THREAD

**Discussion by cbfiddle** [Homebrew/discussions#5609] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.21
ORIGIN: https://github.com/Homebrew/brew
HEAD: c5d09b411776547bdbc8e0435deaf690ca632413
Last commit: 4 days ago
Core tap JSON: 16 Sep 14:33 UTC
Core cask tap JSON: 16 Sep 14:33 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 10-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0-arm64
CLT: 16.0.0.0.1.1723780521
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  wkhtmltopdf

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libwkhtmltox.0.12.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/jnu.h
  /usr/local/include/wkhtmltox/image.h
  /usr/local/include/wkhtmltox/pdf.h

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.profile

Warning: Your Xcode (15.4) at /Applications/Xcode.app is too outdated.
Please update to Xcode 16.0 (or delete it).
Xcode can be updated from the App Store.
```

### Description of issue

When I installed weasyprint, it upgraded mariadb. I'm trying to understand why that happened. The command output says that mariadb is a dependency of an upgraded formula, but I cannot figure out what the dependency is.
I tried `brew deps --include-build --include-test --include-optional weasyprint` but mariadb is not in the list.
I tried `brew uses --eval-all --recursive --include-build --include-test --include-optional mariadb` but it only listed qt-mariadb, which is not installed.
What am I missing?

**Comment by carlocab** [comment]
What probably happened is that installing `weasyprint` upgraded another formula (likely one of `weasyprint`'s dependencies, as shown by `brew deps weasyprint`) that `mariadb` itself depends on. That is, `weasyprint` and `mariadb` share common dependencies. You can view which these are with `brew deps weasyprint mariadb`.

**Reply by cbfiddle** [reply]
Thank you, that explains it. There are two shared dependencies that were updated.
May I suggest that the command output could be clearer on this point?
It said:
```
==> Upgrading 7 dependents of upgraded formulae:
Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
```
The help for `install` is also unclear on this point.
```
Unless HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK is set, brew upgrade or brew
reinstall will be run for outdated dependents and dependents with broken
linkage, respectively.
```
The word "dependents" seems to mean something broader than the standard meaning.

**Reply by carlocab** [reply]
> The word "dependents" seems to mean something broader than the standard meaning.

I don't quite follow.

> ```
> ==> Upgrading 7 dependents of upgraded formulae:
> Disable this behaviour by setting HOMEBREW_NO_INSTALLED_DEPENDENTS_CHECK.
> ```

This seems accurate, given:

> There are two shared dependencies that were updated.

But we're happy to review PRs to make the messaging clearer here.
